src/ast/passes/semantic_analyser.cpp

```diff
@@ -574,6 +574,55 @@ static const std::map<std::string, call_spec> CALL_SPEC = {
           arg_type_spec{ .type=Type::string },
           arg_type_spec{ .type=Type::string },
           arg_type_spec{ .type=Type::integer, .literal=true } } } },
+  { "strcmp",
+    { .min_args=2,
+      .max_args=2,
+      .discard_ret_warn = true,
+      .arg_types={
+          arg_type_spec{ .type=Type::string },
+          arg_type_spec{ .type=Type::string } } } },
+  { "strnlen",
+    { .min_args=2,
+      .max_args=2,
+      .discard_ret_warn = true,
+      .arg_types={
+          arg_type_spec{ .type=Type::string },
+          arg_type_spec{ .type=Type::integer } } } },
+  { "strlen",
+    { .min_args=1,
+      .max_args=1,
+      .discard_ret_warn = true,
+      .arg_types={
+          arg_type_spec{ .type=Type::string } } } },
+  { "strspn",
```

Comment:
Not sure if this brings any value to bpftrace. If so, perhaps we should think about a better name. Same applies to `strcspn`.